src/Paket.Core/Dependencies/NuGetCache.fs

```diff
@@ -454,13 +455,27 @@ let getDetailsFromCacheOr force nugetURL (packageName:PackageName) (version:SemV
         | Some res -> return res
     }
 
+let private pathResolver = NuGet.Packaging.PackagePathResolver(Constants.UserNuGetPackagesFolder, useSideBySidePaths = false)
+let private nugetSettings = { new NuGet.Configuration.ISettings with
+                                    override this.AddOrUpdate(sectionName: string, item: NuGet.Configuration.SettingItem): unit = ()
+                                    override this.GetConfigFilePaths(): Collections.Generic.IList<string> = ResizeArray() :> _
+                                    override this.GetConfigRoots(): Collections.Generic.IList<string> = ResizeArray () :> _
+                                    override this.GetSection(sectionName: string): NuGet.Configuration.SettingSection = null
+                                    override this.Remove(sectionName: string, item: NuGet.Configuration.SettingItem): unit = ()
+                                    override this.SaveToDisk(): unit = ()
+                                    [<CLIEvent>]
+                                    override this.SettingsChanged: IEvent<EventHandler,EventArgs> = Event<EventHandler,EventArgs>().Publish
+                                    }
+let private signingContext = Signing.ClientPolicyContext.GetClientPolicy(nugetSettings, NuGet.Common.NullLogger.Instance)
+let private extractionContext = NuGet.Packaging.PackageExtractionContext(PackageSaveMode.Defaultv3, XmlDocFileSaveMode.Compress, signingContext, NuGet.Common.NullLogger.Instance)
+
 /// Extracts the given package to the user folder
-let rec ExtractPackageToUserFolder(fileName:string, packageName:PackageName, version:SemVerInfo, kind:PackageResolver.ResolvedPackageKind) =
+let rec ExtractPackageToUserFolder(source: PackageSource, fileName:string, packageName:PackageName, version:SemVerInfo, kind:PackageResolver.ResolvedPackageKind) =
```

Comment:
the packagesource had to be added here so that we could get the url from it, because the nuget.packaging APIs expect that as the key that's used for signing verification.

note that we don't do signing verification here, but now we have the scaffolding to do so using nuget's own APIs.  if we modify the `signingContext` above and preload it with hashes from our lockfile for example, then we'd start getting signing validation for free.